utilities/ArgumentParser.cpp

```diff
@@ -58,49 +58,195 @@ std::map<std::string,std::function<bool(const std::string&)>> ArgumentParser::in
+  CHECK_EQ(argParser.getInt("2147483647", parserToTest), 2147483647);
+  CHECK_THROWS(argParser.getInt("2147483648", parserToTest));
+  CHECK_EQ(argParser.getInt("-1", parserToTest), -1);
+  CHECK_EQ(argParser.getInt("-2147483648", parserToTest), -2147483648);
+}
+TEST_CASE("test string parsing") {
+  DoctestArgumentParser argParser;
+  one_bit::ArgumentParser parserToTest;
+  std::vector<string> testStrings{
+    "", 
+    "test", 
+    "0xDEADBEEF", 
+    //"��u�", 
+    "1234", 
+    "string with spaces",
+    //"this is a very long string that exceeds 256 characters, so we'll try and figure out whether the parser can cope with that, but it is very hard to come up with that amount of text, so this is mainly gibberish, and perhaps it's all in vain for parser failure",
+    //"$p�cial!", 
+    "_underscore", 
+  };
+  for (auto& str : testStrings)
+  {
+    CHECK_EQ(argParser.getString(str, parserToTest), str);
+  }
+}
+
+TEST_CASE("test bool parsing") {
+  DoctestArgumentParser argParser;
+  one_bit::ArgumentParser parserToTest;
+  std::vector<string> true_values{ "true", "True", "TRUE", "t", "y", "1" };
+  std::vector<string> false_values{ "false", "False", "FALSE", "f", "n", "0" };
+  std::vector<string> invalid_values{ "fals", "falsee", "Fals", "Falsee", "FALS", "FALSEE", "FaLsE", "falsf", "tru", "truee", "Tru", "Truee", "TRU", "TRUEE", "TrUe", "truf", "w", "r", "00", "01", "2", "-1", "false.", "true." };
+  for (auto& str : true_values)
+  {
+    CHECK(argParser.getBool(str, parserToTest));
+  }
+  for (auto& str : false_values)
+  {
+    CHECK(! argParser.getBool(str, parserToTest));
+  }
+  for (auto& str : invalid_values)
+  {
+    CHECK_THROWS(argParser.getBool(str, parserToTest));
+  }
+}
+
+TEST_CASE("test UiMode parsing") {
+  DoctestArgumentParser argParser;
+  one_bit::ArgumentParser parserToTest;
+  CHECK_EQ(argParser.getUiMode("NONE", parserToTest), one_bit::UiMode::NONE);
+#ifdef USE_QT5
+  CHECK_EQ(argParser.getUiMode("QT", parserToTest), one_bit::UiMode::QT);
+#else
+  CHECK_THROWS(argParser.getUiMode("QT", parserToTest));
+#endif
+  CHECK_THROWS(argParser.getUiMode("Qt", parserToTest));
+  CHECK_THROWS(argParser.getUiMode("CEGUI", parserToTest));
+  CHECK_THROWS(argParser.getUiMode("GTK", parserToTest));
+  CHECK_THROWS(argParser.getUiMode("", parserToTest));
+  CHECK_THROWS(argParser.getUiMode("PINK_ELEPHANT", parserToTest));
+  CHECK_THROWS(argParser.getUiMode("QTGUI", parserToTest));
+  CHECK_THROWS(argParser.getUiMode("NO", parserToTest));
 }
+
+TEST_CASE("test CropRegion parsing") {
+  DoctestArgumentParser argParser;
+  one_bit::ArgumentParser parserToTest;
+  CHECK_EQ(argParser.getCropRegion("TOP_LEFT", parserToTest), one_bit::CropRegion::TOP_LEFT);
+  CHECK_EQ(argParser.getCropRegion("TOP", parserToTest), one_bit::CropRegion::TOP);
+  CHECK_EQ(argParser.getCropRegion("TOP_RIGHT", parserToTest), one_bit::CropRegion::TOP_RIGHT);
+  CHECK_EQ(argParser.getCropRegion("TOPF", parserToTest), one_bit::CropRegion::TOP_LEFT);
```

Comment:
there should be a variable containing TOP_LEFT for those checks that actually don't parse